Server/src/main/java/org/gluu/oxauth/model/registration/RegisterParamsValidator.java

```diff
@@ -306,6 +307,10 @@ public boolean validateRedirectUris(List<GrantType> grantTypes, List<ResponseTyp
             } catch (Exception e) {
                 log.debug(e.getMessage(), e);
                 valid = false;
+            } finally {
+                if (!valid) {
+                    noRedirectUriInSectorIdentifierUri = true;
```

Comment:
Sounds like we can directly throw exception here and remove `noRedirectUriInSectorIdentifierUri` variable.
```
throw errorResponseFactory.createWebApplicationException(Response.Status.BAD_REQUEST, RegisterErrorResponseType.INVALID_CLIENT_METADATA, "Failed to validate redirect uris. No redirect_uri in sector_identifier_uri content.");
```

Reply:
I have moved the changes from Jans to oxAuth.
This is the Jans reference issue: https://github.com/JanssenProject/jans/issues/3639.

Now that we can improve it like this would be ideal.
But we should test the certification again for that case.
If the certification comes out successful, we can take it to Jans as well.